### PR TITLE
fix: four lib/audio_net bugs

### PR DESCRIPTION
**Pull request by rokubop**
A few bug fixes
- `AudioNetTrainer` to use instance variables instead of class variables
- Check folder exists for replays
- Fix a batch size 512 assumption on the last batch
- Update some mean stats which will be used by GUI

### CODE REVIEW

File: lib/audio_net.py [hunk 1]
```diff
@@ -64,25 +64,25 @@ def forward(self, x):
...
-    train_indices = []
     input_size = 120
     
     def __init__(self, dataset, net_count = 1, audio_settings = None):
+        self.nets = []
+        self.optimizers = []
+        self.random_seeds = []
+        self.train_indices = []
+        self.train_loaders = []
+        self.validation_loaders = []
```
Comment (rokubop) [inline lib/audio_net.py:84]: This was fine the first time you train a model, but if you trained a second model in the same session, it was reusing nets, optimizers, etc from the first model.

File: lib/audio_net.py [hunk 2]
```diff
@@ -164,7 +165,7 @@ def train(self, filename):
                             
                             if( i % 10 == 0 ):
                                 correct_in_minibatch = ( local_labels == output.max(dim = 1)[1] ).sum()
-                                print('[Net: %d, %d, %5d] loss: %.3f acc: %.3f' % (j + 1, epoch + 1, i + 1, (running_loss[j] / 10), correct_in_minibatch.item()/self.batch_size))
+                                print('[Net: %d, %d, %5d] loss: %.3f acc: %.3f' % (j + 1, epoch + 1, i + 1, (running_loss[j] / 10), correct_in_minibatch.item()/local_labels.size(0)))
```
Comment (rokubop) [inline lib/audio_net.py:168]: `local_labels.size(0)` should also be `512` for every full batch (same as `self.batch_size`)
But the difference is on the last batch of an epoch which is usually shorter than `512`.  `local_labels.size(0)` should reflect the actual divisor in that case.

File: lib/audio_net.py [hunk 3]
```diff
@@ -229,7 +237,7 @@ def train(self, filename):
                 
                 csv_row = { 'epoch': epoch, 'loss': np.sum(epoch_loss), 'avg_validation_accuracy': np.average(accuracy) }
                 for dataset_label in self.dataset_labels:
-                    csv_row[dataset_label] = accuracy_batch['percent'][dataset_label]
+                    csv_row[dataset_label] = mean_label_accuracy[dataset_label]
```
Comment (rokubop) [inline lib/audio_net.py:240]: this `csv_row` appears to be unused in current cli - but the gui will use these numbers to show live mean accuracy per sound while training.

`accuracy_batch` gets reset for each net, so this line was previously only using last net's numbers instead of the mean

File: lib/audio_net.py (hunk 1 above)
Comment (chaosparrot) [reply lib/audio_net.py:84]: Oh right! Because the initialization was never reset 🤦‍♂️ Good catch, my training runs are usually multiple hours long, and quiting out I tended to do with CTRL+C which quits the current session.

File: lib/audio_net.py (hunk 2 above)
Comment (chaosparrot) [reply lib/audio_net.py:168]: Makes sense!

File: lib/audio_net.py (hunk 3 above)
Comment (chaosparrot) [reply lib/audio_net.py:240]: I think I used this csv file in some dead code I had that I used to make accuracy graphs for each sound as they were training. I think I uncommented that code that is lying somewhere to make those bar charts in mathplotlib.

Agreed that it is better to get the mean rather than the last net, since the last one is rather arbitrary.

**Review by chaosparrot — APPROVED** [review]
